doc/developer-notes.md

```diff
@@ -171,3 +171,36 @@ Threads
 - BitcoinMiner : Generates bitcoins (if wallet is enabled).
 
 - Shutdown : Does an orderly shutdown of everything.
+
+Ignoring IDE/editor files
+--------------------------
+
+In closed-source environments in which everyone uses the same IDE it is common
+to add temporary files it produces to the project-wide `.gitignore` file.
```

Comment:
Also fairly standard practice in open-source environments... (Not saying we have to do it, but this description isn't correct.)